pkg/awsutils/awsutils.go

```diff
@@ -1330,6 +1442,32 @@ func (cache *EC2InstanceMetadataCache) DeallocIPAddresses(eniID string, ips []st
 		log.Errorf("Failed to deallocate a private IP address %v", err)
 		return errors.Wrap(err, fmt.Sprintf("deallocate IP addresses: failed to deallocate private IP addresses: %s", ips))
 	}
+	log.Debugf("Successfully freed IPs %v from ENI %s", ips, eniID)
+	return nil
+}
+
+// DeallocPrefixAddresses frees Prefixes on an ENI
+func (cache *EC2InstanceMetadataCache) DeallocPrefixAddresses(eniID string, prefixes []string) error {
+	if len(prefixes) == 0 {
+		return nil
+	}
+	log.Infof("Trying to unassign the following Prefixes %s from ENI %s", prefixes, eniID)
+	prefixesInput := aws.StringSlice(prefixes)
+
+	input := &ec2.UnassignPrivateIpAddressesInput{
+		NetworkInterfaceId: aws.String(eniID),
+		Ipv4Prefixes:       prefixesInput,
+	}
+
+	start := time.Now()
+	_, err := cache.ec2SVC.UnassignPrivateIpAddressesWithContext(context.Background(), input)
+	awsAPILatency.WithLabelValues("UnassignPrivateIpAddresses", fmt.Sprint(err != nil), awsReqStatus(err)).Observe(msSince(start))
```

Comment:
UnassignPrivateIpAddresses-> DeallocPrefixAddresses

Reply:
Yeah post GA, I can split AssignPrivateIP and UnAssignPrivateIP into prefixes and IPs. Today check is done inside the function since on the input changes.